python/lsst/meas/algorithms/maskStreaks.py

```diff
@@ -0,0 +1,788 @@
+            clusterStandardDeviations = np.zeros((nClusters, 2))
+            for c in range(nClusters):
+                inCluster = X[kmeans.labels_ == c]
+                clusterStandardDeviations[c] = np.std(inCluster, axis=0)
+            # Are the rhos and thetas in each cluster all below the threshold?
+            if (clusterStandardDeviations <= 1).all():
+                break
+            nClusters += 1
+
+        # The cluster centers are final line estimates
+        finalClusters = kmeans.cluster_centers_.T
+
+        # Rescale variables:
+        finalRhos = finalClusters[0] * self.config.rhoBinSize
+        finalThetas = finalClusters[1] * self.config.thetaBinSize
+        result = LineCollection(finalRhos, finalThetas)
+        self.log.info("Lines were grouped into %s potential streak(s)", len(finalRhos))
+
+        return result
+
+    def _fitProfile(self, lines, maskedImage):
+        """Fit the profile of the streak.
+
+        Given the initial parameters of detected lines, fit a model for the
+        streak to the original (non-binary image). The assumed model is a
+        straight line with a Moffat profile.
+
+        Parameters
+        ----------
+        lines : `LineCollection`
+            Collection of guesses for `Line`s detected in the image.
+        maskedImage : `lsst.afw.image.maskedImage`
+            Original image to be used to fit profile of streak.
+
+        Returns
+        -------
+        lineFits : `LineCollection`
+            Collection of `Line` profiles fit to the data.
+        finalMask : `np.ndarray`
+            2d mask array with detected streaks=1.
+        """
+        data = maskedImage.image.array
+        weights = maskedImage.variance.array**-1
+        # Mask out any pixels with non-finite weights
+        weights[~np.isfinite(weights) | ~np.isfinite(data)] = 0
+
+        lineFits = LineCollection([], [])
+        finalLineMasks = [np.zeros(data.shape, dtype=bool)]
+        nFinalLines = 0
+        for line in lines:
+            line.sigma = self.config.invSigma**-1
+            lineModel = LineProfile(data, weights, line=line)
+            # Skip any lines that do not cover any data (sometimes happens because of chip gaps)
+            if lineModel.lineMaskSize == 0:
+                continue
+
+            fit, chi2, fitFailure = lineModel.fit(dChi2Tol=self.config.dChi2Tolerance, log=self.log)
+            if fitFailure:
+                self.log.warning("Streak fit failed.")
+
+            # Initial estimate should be quite close: fit is deemed unsuccessful if rho or theta
+            # change more than the allowed bin in rho or theta:
+            if ((abs(fit.rho - line.rho) > 2 * self.config.rhoBinSize)
+                    or (abs(fit.theta - line.theta) > 2 * self.config.thetaBinSize)):
+                fitFailure = True
+                self.log.warning("Streak fit moved too far from initial estimate. Line will be dropped.")
```

Comment:
As above, please add some identifier for which line is being dropped.

Reply:
Same as above--I don't want to put any specifics here.

Reply:
As above, these fit failures can be counted either separately or together with the previous failures, and just issue a warning after the loop over `lines`